File: PolinomioArreglo.c
#include <stdio.h>

int main(int argc, char *argv[]) {
	
	int P1[50],P2[50],P3[50];
	int Grado_1,Grado_2;
	int mayor,DG,i;
	
	printf("= Polinomio 1 =\n\tDe que grado es? = ");
	scanf("%d",&Grado_1);
	printf("\t Ingresa los coeficientes:\n");
	for(i=0;i<=Grado_1;i++){
		printf("\t\t [%d] : ",i+1);
		scanf("%d",&P1[i]);
	}
	
	printf("= Polinomio 2 =\n\tDe que grado es? = ");
	scanf("%d",&Grado_2);
	printf("\t Ingresa los coeficientes:\n");;
	for(i=0;i<=Grado_2;i++){
		printf("\t\t [%d] : ",i+1);
		scanf("%d",&P2[i]);
	}
	
	if(Grado_2<Grado_1){
		DG = Grado_1 - Grado_2;
		mayor = Grado_1;
		for(i=0;i<DG;i++){		
			P3[i]=P1[i];
		}
		for(i=DG;i<=Grado_1;i++){
			P3[i] = P1[i] + P2[i-DG];
		}
	}else if(Grado_1<Grado_2){
		DG = Grado_2 - Grado_1;
		mayor = Grado_2;
		for(i=0;i<DG;i++){		
			P3[i]=P2[i];
		}
		for(i=DG;i<=Grado_2;i++){
			P3[i] = P1[i-DG] + P2[i];
		}
		
	}else{
		mayor = Grado_1;
		for(i=0;i<=Grado_1;i++){
			P3[i] = P1[i] + P2[i];
		}
	}
	
	printf("El resultado es: ");
	for(i=0;i<mayor;i++){
		printf("%dx^%d + ",P3[i],mayor-i);
	}
	printf("%d",P3[mayor]);

	return 0;
}

File: PolinomioArreglo.scala
object Polinomio{

    def main (args: Array[String]){
      var P1 = new Array[Int](50)
      var P2 = new Array[Int](50)
      var PM = new Array[Int](50)
      var menor = 0
      var mayor = 0
      var grado_1 = 0
      var grado_2 = 0
        
        println ("= Polinomio 1 =")
        println ("De que grado es?")
        grado_1 = readInt()
        println ("Ingresa los coeficientes:")
        for(i<-0 to grado_1){
            P1(i) = readInt()
        }

        println ("\n= Polinomio 2 =")
        println ("De que grado es?")
        grado_2 = readInt()
        println ("Ingresa los coeficientes:") 
        for(i<-0 to grado_2){
            P2(i) = readInt()
        }

        if (grado_1>=grado_2){
        PM=P1
        menor=grado_2
        mayor=grado_1

        }
        else {
        PM=P1
        menor=grado_1
        mayor=grado_2
        }
        for(i<-0 to menor){
            PM(i) = P1(i) + P2(i)
        }
        for(i<-menor + 1 to mayor){
        PM(i) = PM(i)
        }
    
    println("El resultado de la suma es:")
        for(i<-0 to mayor) {
            println(+ PM(i) + "x^" + i)
        }

    }


}